src/test/kotlin/se/uulm/snowballr/backend/service/paper/GetBackwardReferencedPapersTest.kt

```diff
@@ -0,0 +1,131 @@
+        Arguments.of(citationRepoMock::getBackwardsReferencedPaperIdsOfPaperById),
+        Arguments.of(authorOfPaperRepoMock::getAuthorsOfPaperById),
+        Arguments.of(citationRepoMock::getBackwardsReferencedPaperIdsOfPaperById),
+    )
+
+    @Suppress("ReturnCount")
+    private fun mockHappyPathUntil(failAt: KFunction<*>?) {
+        val paper = DataBuilder.createExamplePaper(id = requestId)
+        val backwardReferenceId = UUID.randomUUID()
+        val referencedPaper = DataBuilder.createExamplePaper(id = backwardReferenceId)
+        val author = DataBuilder.createExampleAuthor()
+
+        if (failAt == paperRepoMock::doesPaperExistById) {
+            coEvery { paperRepoMock.doesPaperExistById(paper.id) } throws TestSpecificException()
+            return
+        }
+        coEvery { paperRepoMock.doesPaperExistById(paper.id) } returns true
+
+        if (failAt == citationRepoMock::getBackwardsReferencedPaperIdsOfPaperById) {
+            coEvery {
+                citationRepoMock.getBackwardsReferencedPaperIdsOfPaperById(paper.id)
+            } throws TestSpecificException()
+            return
+        }
+        coEvery {
+            citationRepoMock.getBackwardsReferencedPaperIdsOfPaperById(paper.id)
+        } returns listOf(backwardReferenceId)
+
+        if (failAt == paperRepoMock::getPaperById) {
+            coEvery { paperRepoMock.getPaperById(backwardReferenceId) } throws TestSpecificException()
+            return
+        }
+        coEvery { paperRepoMock.getPaperById(backwardReferenceId) } returns referencedPaper
+
+        if (failAt == authorOfPaperRepoMock::getAuthorsOfPaperById) {
+            coEvery { authorOfPaperRepoMock.getAuthorsOfPaperById(backwardReferenceId) } throws TestSpecificException()
+            return
+        }
+        coEvery { authorOfPaperRepoMock.getAuthorsOfPaperById(backwardReferenceId) } returns listOf(author)
+
+        if (failAt == citationRepoMock::getBackwardsReferencedPaperIdsOfPaperById) {
+            coEvery {
+                citationRepoMock.getBackwardsReferencedPaperIdsOfPaperById(backwardReferenceId)
+            } throws TestSpecificException()
+            return
+        }
+        coEvery {
+            citationRepoMock.getBackwardsReferencedPaperIdsOfPaperById(backwardReferenceId)
+        } returns listOf(UUID.randomUUID())
+    }
+
+    @ParameterizedTest
+    @MethodSource("failingFunctions")
+    fun `When a step fails, then an exception is thrown`(failAt: KFunction<*>) = runTest {
+        mockHappyPathUntil(failAt)
+        assertThrows<TestSpecificException> {
+            mainService.getBackwardReferencedPapers(getExampleRequest())
+        }
+    }
+
+    @Test
+    fun `When the paper doesn't exist, then a NotFoundException is thrown`() = runTest {
+        coEvery { paperRepoMock.doesPaperExistById(requestId) } throws SnowballRException.NotFoundException(
+            entityType = EntityType.PAPER, requestId.toString(),
+        )
```

Comment:
```suggestion
        coEvery { paperRepoMock.doesPaperExistById(requestId) } returns false
```

One little thing I found, sorry.

Reply:
Oh shit, good catch. I forgot to change this. Thanks.